src/main/java/app/controllers/PresentationController.java

```diff
@@ -39,12 +56,59 @@ public String presentation(@PathVariable Long id,
         }
         return times;
     }
+    @PostMapping("/project/{pid}/presentation/update")
+    public String updateTimeSlots(@PathVariable Long pid,
+                                  @RequestParam("timeSlot") String id,
+                                  @AuthenticationPrincipal UserDetails currentUser) {
+
+        System.out.println();
+        System.out.println();
+        System.out.println();
+        System.out.println();
+        if(id.split(",").length > 1) {
+            System.out.println("skipping");
+            return "redirect:/project/" + pid + "/presentation";
+        }
+        AuthenticatedUser authenticatedUser = authenticatedUserService.findByUsername(currentUser.getUsername());
+        User user;
+        Project proj = projectRepository.findOne(pid);
+        if(authenticatedUser.getType().equals("Student")) {
+            System.out.println("student");
+            user = authenticatedUser.getStudent();
+            Student stud = (Student) user;
```

Comment:
Similar to the other controller action, `authenticatedUser.getStudent()` returns a student already, no need to cast since it's only used inside of this scope.